hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INodeDirectory.java

```diff
@@ -572,6 +572,25 @@ public boolean addChild(INode node) {
     return true;
   }
 
+  /**
+   * During image loading, the search is unnecessary since the insert position
+   * should always be at the end of the map given the sequence they are
+   * serialized on disk.
+   */
+  public boolean addChildAtLoading(INode node) {
+    int pos;
+    if (!node.isReference()) {
+      pos = (children == null) ? (-1) : (-children.size() - 1);
+    } else {
+      pos = searchChildren(node.getLocalNameBytes());
```

Comment:
can we reuse addChild() here to make it cleaner?
My understanding is that in the case of INodeReference, we fall back to the original behavior.

That is, 
`
} else {
  return addChild(node);
}
`